File: minSubArrayLenSlidingWindow.go
func MinSubArrayLenSlidingWindow(target int, nums []int) int {
	left, right := 0, 0
	currentSum := 0
	result := math.MaxInt32
	n := len(nums)

	for right < n {
		currentSum += nums[length]

		for currentSum >= target && left <= right {
			result = min(result, right - left + 1)
			currentSum -= nums[left]
			left++
		}

		right++
	}

	if result == math.MaxInt32 {
		return 0
	}

	return result
}

File: minSubArrayLenSlidingWindow.py
def minSubArrayLenSlidingWindow(target: int, nums: list[int]) -> int:
    left = right = 0
    current_sum = 0
    result = float('inf')
    n = len(nums)

    while right < n:
        current_sum += nums[right]
        while current_sum >= target:
            result = min(result, right - left + 1)
            current_sum -= nums[left]
            left += 1
        right += 1
    return result if result != float('inf') else 0